backend_windows.go

```diff
@@ -638,16 +648,8 @@ func (w *Watcher) readEvents() {
 
 			// Point "raw" to the event in the buffer
 			raw := (*windows.FileNotifyInformation)(unsafe.Pointer(&watch.buf[offset]))
-
-			// Create a buf that is the size of the path name
-			size := int(raw.FileNameLength / 2)
-			var buf []uint16
-			// TODO: Use unsafe.Slice in Go 1.17; https://stackoverflow.com/questions/51187973
-			sh := (*reflect.SliceHeader)(unsafe.Pointer(&buf))
-			sh.Data = uintptr(unsafe.Pointer(&raw.FileName))
-			sh.Len = size
-			sh.Cap = size
-			name := windows.UTF16ToString(buf)
+			buf := unsafe.Slice(&raw.FileName, raw.FileNameLength/2)
+			name := syscall.UTF16ToString(buf)
```

Comment:
[Question]
Why the change from `windows` to `syscall`?

Reply:
Again, not very satisfying.  But this what the original code does.

https://github.com/adriansr/fsnotify/blob/a81f2b630e7c708d59a259e6308cd70409a1ebd3/windows.go#L464-L465

Reply:
I'm ok with changing this later, but I think we should start with a copy of the code we are currently shipping.